File: biblioteca.java
public class Livro {
    private String titulo;
    private String autor;
    private int anoPublicacao;
    private int numCopiasDisponiveis;

    public Livro(String titulo, String autor, int anoPublicacao, int numCopiasDisponiveis) {
    }

    public boolean emprestar() {
        return false;
    }

    public void devolver() {
    }

    public boolean copiasDisponiveis() {
        return false;
    }

    public String obterInformacoes() {
        return null;
    }
}

File: biblioteca.hs
type Titulo = String
type Autor = String
type AnoPublicacao = Int
type NumCopiasDisponiveis = Int

data Livro = Livro Titulo Autor AnoPublicacao NumCopiasDisponiveis deriving (Show)

emprestar :: Livro -> (Bool, Livro)
emprestar = undefined

devolver :: Livro -> Livro
devolver = undefined

copiasDisponiveis :: Livro -> Bool
copiasDisponiveis = undefined

obterInformacoes :: Livro -> String
obterInformacoes = undefined
